generators/typescript/sdk/client-class-generator/src/endpoint-request/GeneratedBytesEndpointRequest.ts

```diff
@@ -0,0 +1,231 @@
+                return undefined;
+            }
+        }
+
+        return result;
+    }
+
+    public getEndpointParameters(context: SdkContext): OptionalKind<ParameterDeclarationStructure>[] {
+        const parameters: OptionalKind<ParameterDeclarationStructure>[] = [
+            {
+                name: GeneratedBytesEndpointRequest.BYTES_VARIABLE_NAME,
+                type: getTextOfTsNode(this.getFileParameterType(context))
+            }
+        ];
+
+        for (const pathParameter of getPathParametersForEndpointSignature(this.service, this.endpoint)) {
+            parameters.push({
+                name: getParameterNameForPathParameter({
+                    pathParameter,
+                    retainOriginalCasing: this.retainOriginalCasing
+                }),
+                type: getTextOfTsNode(context.type.getReferenceToType(pathParameter.valueType).typeNode)
+            });
+        }
+
+        if (this.requestParameter != null) {
+            parameters.push(this.requestParameter.getParameterDeclaration(context));
+        }
+
+        return parameters;
+    }
+
+    private getFileParameterType(context: SdkContext): ts.TypeNode {
+        const types: ts.TypeNode[] = [ts.factory.createTypeReferenceNode("File")];
+
+        visitJavaScriptRuntime(this.targetRuntime, {
+            node: () => {
+                types.push(context.externalDependencies.fs.ReadStream._getReferenceToType());
+
+                types.push(context.externalDependencies.blob.Blob._getReferenceToType());
+            },
+            browser: () => {
+                types.push(ts.factory.createTypeReferenceNode("Blob"));
+            }
+        });
+
+        if (this.requestBody.isOptional) {
+            types.push(ts.factory.createKeywordTypeNode(ts.SyntaxKind.UndefinedKeyword));
+        }
+
+        return ts.factory.createUnionTypeNode(types);
+    }
+
+    public getBuildRequestStatements(context: SdkContext): ts.Statement[] {
+        const statements: ts.Statement[] = [];
+
+        if (this.requestParameter != null) {
+            statements.push(...this.requestParameter.getInitialStatements());
+        }
+
+        if (this.queryParams != null) {
+            statements.push(...this.queryParams.getBuildStatements(context));
+        }
+
+        return statements;
+    }
+
+    public getFetcherRequestArgs(
+        context: SdkContext
+    ): Pick<Fetcher.Args, "headers" | "queryParameters" | "body" | "contentType" | "requestType" | "duplex"> {
+        return {
+            headers: this.getHeaders(context),
+            queryParameters: this.queryParams != null ? this.queryParams.getReferenceTo(context) : undefined,
+            contentType: this.requestBody.contentType,
+            requestType: "bytes",
+            body: ts.factory.createIdentifier(GeneratedBytesEndpointRequest.BYTES_VARIABLE_NAME),
+            duplex: ts.factory.createStringLiteral("half")
```

Comment:
nit: why does this have tsmorph?

Reply:
`duplex` is expecting a ts-morph expression! I couldn't handled it within the function, but seemed roughly the same complexity, so just made a ts-morph string literal